### ISSUE TITLE
New Version 8.0 not always closing correctly on Ubuntu

### ISSUE THREAD

**Issue by PeterNJackson** [racket/racket#3688]
If I open the version 8.0, enter some simple code (+ 3 2) , run it and then close the IDE (press X top right corner). The "gracket" process doesn't end. So I can't restart the IDE. But if I use "Quit" from the file menu everything closes fine.

**Comment by mflatt** [comment]
Thanks for the report! But I'm not able to replicate this, yet. Can you say a little more about which Ubuntu version you're running and how you installed and started DrRacket?

**Comment by PeterNJackson** [comment]
Hi Mathew

I am running Ubuntu 20.04. Racket was installed as an upgrade from 
Racket 7.9 which was originally upgraded from 7.8.

It is launched from:

[Desktop Entry]
Name=DrRacket
GenericName=IDE for Racket
Comment=DrRacket is an interactive, integrated, graphical programming 
environment for the Racket programming languages.
Terminal=false
Type=Application
Categories=Development;Education;
Exec=/usr/bin/drracket -singleInstance %F
Icon=/usr/share/racket/drracket-exe-icon.png

**Comment by mflatt** [comment]
Hm... still no luck.

Do you have a `~/.racket` directory? Or `~/.local/share/racket`? If so, does temporarily renaming it change anything?

**Comment by PeterNJackson** [comment]
I did have a ~/.local/share/racket directory.  But renaming it doesn't seem to change anything. Peter Sent from my Galaxy
-------- Original message --------From: Matthew Flatt <notifications@github.com> Date: 2021-02-17  6:22 p.m.  (GMT-08:00) To: racket/racket <racket@noreply.github.com> Cc: PeterNJackson <Peter.jackson@ymail.com>, Author <author@noreply.github.com> Subject: Re: [racket/racket] New Version 8.0 not always closing correctly on Ubuntu (#3688) 
Hm... still no luck.
Do you have a ~/.racket directory? Or ~/.local/share/racket? If so, does temporarily renaming it change anything?

**Comment by mflatt** [comment]
If you run `drracket` from the command line, does it behave the same way? That is, after you close the DrRacket window, the command prompt doesn't return?

In that case, there's some chance that hitting Ctl-C will provide useful output from DrRacket on why it hasn't exited.

**Comment by PeterNJackson** [comment]
peter@peter-HP-PAVILION-NOTEBOOK-15:~$ drracket
^Cuser break
   context...:
/usr/share/racket/pkgs/gui-lib/mred/private/wx/common/queue.rkt:634:3
peter@peter-HP-PAVILION-NOTEBOOK-15:~$

Above is what I received after hitting Ctl-C.

**Comment by PeterNJackson** [comment]
I assume no luck finding the issue. It is happening on both of my 
machines a desktop (an i9 20 core) and my laptop (an i7 8 core).

**Comment by mflatt** [comment]
What happens when you run this simple program?

```
#lang racket/gui
(send (new (class frame%
             (define/augride (on-close)
               (printf "closing\n"))
             (super-new [label "Test"]
                        [width 300]
                        [height 300])))
      show #t)
```

That is, if the above content is in `/tmp/f.rkt` and you run `racket /tmp/f.rkt`? Does it print "closing"? Does the Racket process exit?

This experiment is meant to check whether it's something specific to DrRacket, whether close events reach callbacks, etc.

**Comment by PeterNJackson** [comment]
Yes it prints "closing"If I run it from the command it work. But even when I run it from the ide it works correctly. The problem reappear it I switch back to the student language.  So the issue seems to have something to do with student languages.Peter Sent from my Galaxy
-------- Original message --------From: Matthew Flatt <notifications@github.com> Date: 2021-02-26  10:01 a.m.  (GMT-08:00) To: racket/racket <racket@noreply.github.com> Cc: PeterNJackson <Peter.jackson@ymail.com>, Author <author@noreply.github.com> Subject: Re: [racket/racket] New Version 8.0 not always closing correctly on Ubuntu (#3688) 
What happens when you run this simple program?
#lang racket/gui
(send (new (class frame%
             (define/augride (on-close)
               (printf "closing\n"))
             (super-new [label "Test"]
                        [width 300]
                        [height 300])))
      show #t)

That is, if the above content is in /tmp/f.rkt and you run racket /tmp/f.rkt? Does it print "closing"? Does the Racket process exit?
This experiment is meant to check whether it's something specific to DrRacket, whether close events reach callbacks, etc.

**Comment by mflatt** [comment]
Aha! Yes, I'm able to replicate the problem if I hit "Run" in "Beginner Student".
